mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -299,3 +351,49 @@ func newKubernetesClient(kubeconfigPath string, env adapter.Env) (k8s.Interface,
 	}
 	return k8s.NewForConfig(config)
 }
+
+func (b *builder) createCacheController(k8sInterface k8s.Interface, clusterID string) error {
+	controller, err := getNewCacheController(b, k8sInterface, b.kubeHandler.env)
+	if err == nil {
+		b.controllers[clusterID] = controller
+		b.kubeHandler.k8sCache[clusterID] = controller
+		b.kubeHandler.env.Logger().Infof("created remote controller %s", clusterID)
+	} else {
+		b.kubeHandler.env.Logger().Errorf("error on creating remote controller %s err = %v", clusterID, err)
+	}
+
+	return err
+}
+
+func (b *builder) deleteCacheController(clusterID string) error {
+	b.kubeHandler.k8sCache[clusterID].StopControlChannel()
```

Comment:
I think you want to stop the control Channel which will stop the controllers after you have removed the controller from the implementation.  IOW move this line below line 371.

Reply:
OK. That was a bad idea. Can't call the StopControlChannel() function after deleting the resource that's pointing to the function, i.e. delete(b.kubeHandler.k8sCache, clusterID).